frontend/wasm/src/editor_api.rs

```diff
@@ -435,3 +435,23 @@ impl Drop for JsEditorHandle {
 		EDITOR_INSTANCES.with(|instances| instances.borrow_mut().remove(&self.editor_id));
 	}
 }
+
+#[cfg(test)]
+mod layer_delete_bug_tests {
+	use super::*;
+
+	use editor::misc::test_utils::EditorTestUtils;
+	use editor::viewport_tools::tool::ToolType;
+
+	#[test]
+	fn should_not_crash_when_layer_is_deleted() {
```

Comment:
This should probably say something about "when using a tool", maybe `layer_deleted_while_using_tool` is a better name for it?